packages/binding-http/src/http-server.ts

```diff
@@ -253,8 +252,10 @@ export default class HttpServer implements ProtocolServer {
 
     /** returns server port number and indicates that server is running when larger than -1  */
     public getPort(): number {
-        if (this.server.address() && typeof this.server.address() === "object") {
-            return (<AddressInfo>this.server.address()).port;
+        const address: AddressInfo | string | null = this.server?.address?.();
+
+        if (typeof address === "object") {
+            return address?.port ?? -1;
```

Comment:
Noticed that this getPort method is different from the http-client-basic-test, can you explain why?

Reply:
Very good point, I copied over the code from there to here :)

In general, though, these getters should be revisited later IMHO as returning `-1` does not seem like a very good practice to me. Maybe that is why I forgot to also provide a "proper" fix here :/

Another thing I noticed, by the way, is that we should consider using getters and setters as defined by JavaScript for this kind of method in the future (see, e.g., [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) for an example) which is more idiomatic and simplifies some of the APIs.

The method implementation would then become something like this:

```ts
    class HttpServer {
        public get port(): number {
            return this.#port;
        }
    }
```

While the respective properties could be accessed like so (effectively getting rid of the `get` prefix):

```ts
    const serverPort = httpServer.port;
```

Reply:
+1 for both considerations. However, I would address those in a separate PR. Regarding this PR we can consider this issue resolved.